client/ayon_core/hosts/substancepainter/api/lib.py

```diff
@@ -624,7 +623,8 @@ def _setup_prompt():
                 f"{mesh_filepath}\n\n"
                 "Creating new project directly with the mesh path instead.")
         else:
-            dialog.done(dialog.Accepted)
+            if not allow_user_setting:
+                dialog.done(dialog.Accepted)
```

Comment:
I feel like if we do not allow user settings is there still a reason to go through the convoluted steps of menu entries above, etc. instead of just a "create project" call without dialogs?

Reply:
I understand - but isn't this particular toggle intended to disable that prompt?

Reply:
> I understand - but isn't this particular toggle intended to disable that prompt?

maybe this is not a good naming of variable for this toggle. If "allow_user_setting" is disabled, the system would take preset and click ok to the dialog after loading the mesh. If "allow_user_setting" is enabled, the dialog would be display and user can set the setting after loading the mesh.

Reply:
Exactly. So if this option is False - there is not reason for the dialog to come up to begin with. We could just use substance's API to create the project _without_ even trying with a workaround to bring up the dialog.

Reply:
> Exactly. So if this option is False - there is not reason for the dialog to come up to begin with. We could just use substance's API to create the project _without_ even trying with a workaround to bring up the dialog.

It is by default True, this PR is to make the dialog not disappearing.

Reply:
> The option is by default True, this PR is to make the dialog not disappearing, we still need the dialog so that the user can configure the setting and click ok to create project

Let me rephrase:

If there's any options that "disappears" the dialog as if the user should not interact with it. Then we should just _NOT_ show the dialog to begin with instead of popping up the dialog and accepting it. So if the default `True` means "do not allow the  user to interact with a dialog" then we should just not open the dialog but instead use the api to create the project so we don't need to go through the UI. Yes?

Reply:
the default `True` means "it allows the user to interact with a dialog and they can click ok to create project"